addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -26,6 +26,23 @@ const SharedPopupWidget = publicWidget.Widget.extend({
         'hidden.bs.modal': '_onModalHidden',
     },
 
+    /**
+     * @override
+     */
+    start() {
+        this._super(...arguments);
+
+        if (!this._isNormalCase()) {
+            return;
+        }
+
+        // Refreshing the widget destroys it and gives it the "d-none" class
+        // while the modal is open (e.g. Undo with "Ctrl+Z").
+        if (this.el.firstElementChild.classList.contains('show')) {
+            this.el.classList.remove('d-none');
+        }
```

Comment:
If that comment is true, then the comment below is not:
```javascript
// Popup are always closed when entering/leaving edit mode (see
// PopupWidget), this allows to make sure the class is sync on the
// .s_popup parent after that moment too.
this.el.classList.add('d-none');
```

Also your commit message seems to say the bug is because of a widget refresh when we type text... normally we don't refresh widgets when typing text? What code does that? :open_mouth:

Reply:
> Also your commit message seems to say the bug is because of a widget refresh when we type text... normally we don't refresh widgets when typing text? What code does that? open_mouth

The widget is refreshed when editing text at several times. For example, when doing Undo or during a line break. This comes from the following code in `snippets.editor.js`:
```js
this.options.wysiwyg.odooEditor.addEventListener('historyRevert', _.debounce(() => {
    this.trigger_up('widgets_start_request', {
        $target: this.options.wysiwyg.$editable,
        editableMode: true,
    });
}, 50));
```

Reply:
@bvr-odoo Maybe scary (I'll keep that in mind and investigate once possible) but in the end, it's probably what is done in destroy that's not right in the end: indeed even if "popup are always closed when enterering/leaving edit mode" is true, we cannot rely on destroy being only called at that times. And I don't think we can rely on popup being closed when destroying related widgets either? I think the only code closing popups are the invisibility-button related code?

So probably needs to find another rightful place to add the d-none class when leaving edit mode and that we cannot rely on the `_onModalHidden` handler? :thinking: Are we not waiting for modal to be closed before destroying their related widgets?

Reply:
@qsm-odoo In edit mode, we can rely on the "onTargetHide" method, which is called in "cleanForSave". However, outside of edit mode, the only option is to do it in the "destroy", and there it won't be a problem. Another, perhaps more robust solution, would be to listen for class changes on the "modal" using a "mutationObserver" but I'm not sure it's worth constantly listening for changes in the "s_popup" modal classlist.

Reply:
@bvr-odoo What code does hide popups outside of edit mode (when entering edit mode) ?

Reply:
@qsm-odoo  It is done by the "destroy" method of the "PopupWidget" since this [commit](https://github.com/odoo/odoo/commit/a7ee7f7cb42ab1f89add6651badbc51735f7baac)

Reply:
I made the changes. I'm still wondering if I need to do the same thing that is done in "_isNormalCase" for the new line in the popup options. I don't really understand the utility of it.